FreeRTOS/FreeRTOS
# Sync OTA library from AFR 202007.00 release to lts-development

Description
-----------

Sync OTA library from the AFR release-candidate branch, https://github.com/aws/amazon-freertos/tree/release-candidate. Everything under `FreeRTOS-Plus/Source/FreeRTOS-IoT-Libraries/c_sdk/aws/ota` is in sync with `amazon-freertos` repo.

Major features:

- 2 new API for suspend and resume.
- Now allows downgrade with a user configuration.

Test Steps
-----------

- Demos verified manually.
- Unit tests run and passed from a different repo.

Related Issue
-----------



By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
